dataactcore/scripts/pullFPDSData.py

```diff
@@ -0,0 +1,1158 @@
+    # if a date that the script was last successfully run is provided, we're inserting over something that already
+    # exists so we have to check for conflicts and update when there is one
+    else:
+        for value in data:
+            if i % 5000 == 0:
+                logger.info('inserting row %s for current batch', i)
+
+            tmp_obj = process_data(value['content'][contract_type], atom_type=contract_type, sess=sess)
+            insert_statement = insert(DetachedAwardProcurement).values(**tmp_obj).\
+                on_conflict_do_update(index_elements=['detached_award_proc_unique'], set_=tmp_obj)
+            sess.execute(insert_statement)
+            i += 1
+
+
+def get_data(contract_type, award_type, now, sess, last_run=None):
+    """ get the data from the atom feed based on contract/award type and the last time the script was run """
+    data = []
+    yesterday = now - datetime.timedelta(days=1)
+    log_interval = 1000
+    # if a date that the script was last successfully run is not provided, get all data
+    if not last_run:
+        # we want to log less often for a "get all" run
+        log_interval = 10000
+        # params = 'SIGNED_DATE:[2015/10/01,'+ yesterday.strftime('%Y/%m/%d') + '] '
+        params = 'SIGNED_DATE:[2016/10/01,' + yesterday.strftime('%Y/%m/%d') + '] '
+        # params = 'SIGNED_DATE:[2017/03/01,'+ yesterday.strftime('%Y/%m/%d') + '] '
+    # if a date that the script was last successfully run is provided, get data since that date
+    else:
+        last_run_date = last_run.update_date
+        params = 'LAST_MOD_DATE:[' + last_run_date.strftime('%Y/%m/%d') + ',' + yesterday.strftime('%Y/%m/%d') + '] '
+
+    # TODO remove this later, this is just for testing
+    # params += 'CONTRACTING_AGENCY_ID:1542 '
+    # params = 'VENDOR_ADDRESS_COUNTRY_CODE:"GBR"'
+    # params = 'PIID:"0046"+REF_IDV_PIID:"W56KGZ15A6000"'
+
+    i = 0
+    logger.info('Starting: ' + feed_url + params + 'CONTRACT_TYPE:"' + contract_type.upper() +
+                '" AWARD_TYPE:"' + award_type + '"')
+    while True:
+        resp = requests.get(feed_url + params + 'CONTRACT_TYPE:"' + contract_type.upper() + '" AWARD_TYPE:"' +
+                            award_type + '"&start=' + str(i), timeout=60)
+        resp_data = xmltodict.parse(resp.text, process_namespaces=True,
+                                    namespaces={'http://www.fpdsng.com/FPDS': None,
+                                                'http://www.w3.org/2005/Atom': None})
+        # only list the data if there's data to list
+        try:
+            listed_data = list_data(resp_data['feed']['entry'])
+        except KeyError:
+            listed_data = []
+
+        for ld in listed_data:
+            data.append(ld)
+            i += 1
+
+        # Log which one we're on so we can keep track of how far we are (different numbers for different pulls)
+        if i % log_interval == 0:
+            logger.info("On line " + str(i))
```

Comment:
This log output is the same as line 1062 (logging lines in get_delete_data). Can we specify which we're in to avoid confusion?